Libraries/Network/RCTNetworking.m

```diff
@@ -331,16 +340,47 @@ + (NSString *)decodeTextData:(NSData *)data fromResponse:(NSURLResponse *)respon
   // Attempt to decode text
   NSString *encodedResponse = [[NSString alloc] initWithData:data encoding:encoding];
   if (!encodedResponse && data.length) {
+    
+    
+    
     // We don't have an encoding, or the encoding is incorrect, so now we
     // try to guess (unfortunately, this feature is available in iOS 8+ only)
     if ([NSString respondsToSelector:@selector(stringEncodingForData:
                                                encodingOptions:
                                                convertedString:
                                                usedLossyConversion:)]) {
-      [NSString stringEncodingForData:data
-                      encodingOptions:nil
-                      convertedString:&encodedResponse
-                  usedLossyConversion:NULL];
+      
+      encoding = [NSString stringEncodingForData:data
+                                 encodingOptions:nil
+                                 convertedString:&encodedResponse
+                             usedLossyConversion:NULL];
+      
+      if (encoding > 0 && isIncremental) {
+        // When response data contains multi-byte charactor, incremental data may contain a part of the charactor.
+        // eg. The UTF-8-encoded of `𝟘` is `\xF0\x9D\x9F\x98`, the incremental data may end with `\xF0\x9D`,
```

Comment:
I think the code will work for encodings other than UTF-8, it will try to decode the data first, get the part of string that can be decoded, then the rest of the data should belongs to the character data that hasn't be fully received. So it's not specified to work with only UTF-8.